shared-libs/cht-datasource/src/libs/doc.ts

```diff
@@ -0,0 +1,24 @@
+import { hasField, hasFields, isRecord } from './core';
+
+const isDoc = (value: unknown): value is Doc => isRecord(value) && hasFields(value, [
+  { name: '_id', type: 'string' },
+  { name: '_rev', type: 'string' }
+]);
+
+export const getDocsByIds = (db: PouchDB.Database<Doc>) => async (ids: NonEmptyArray<string>): Promise<Doc[]> => {
+  const results = await db.allDocs({ keys: ids, include_docs: true });
+  return results.rows
+    .map(row => row.doc)
+    .filter(isDoc);
+};
+
+export const getDocById = (db: PouchDB.Database<Doc>) => async (uuid: string): Promise<Nullable<Doc>> => {
+  return (await getDocsByIds(db)([uuid]))[0];
```

Comment:
Honestly I originally used `allDocs` just because I was copying code from `shared-libs/lineage` and that function was handling the case of grabbing multiple docs.  Switched this back up to use `get`.